### PR TITLE
Upgrading ingest-attachment dependencies

### PR DESCRIPTION
**Pull request by kartg**
Signed-off-by: Kartik Ganesh <gkart@amazon.com>

### Description
Multiple dependencies under ingest-attachment have been upgraded:
* Tika libraries upgraded from 1.24.1 to 2.4.0
  * The major version upgrade requires an explicit dependency on `tika-parsers-standard-package` to import the parser implementations, and an update to the namespace of `RTFParser`. 
  * This upgrade also required an update of Apache Commons-IO from 2.7 to 2.11.0, and PDFBox to 2.0.25 as per Tika release notes
  * Also, `LanguageIdentifier` has been deprecated. This must be replaced by a concrete implementation of `LanguageDetector`. Tika publishes an implementation based on Optimaize via `tika-langdetect-optimaize`
    * This in turn brings in dependencies on Optimaize's `language-detector` and Google Guava (set to version 18 since that is what Optimaize uses, and to minimize the list of ignored violations)
    * Language-detector and Guava do not supply LICENSE and NOTICE files in the right format, so these have been manually added
*  xmlbeans libraries updated from 3.0.1 to 5.0.2
   * xmlbeans is now a subproject of Apache POI, so the POI libraries were upgraded from 4.1.2 to 5.2.2
   * With POI 5.x the `ooxml-schemas` library has been moved to `ooxml-lite` / `ooxml-full`. Since `ooxml-schemas` no longer exists, the LICENSE and NOTICE files in the `licenses/` directory have been removed.
 
Alongside these version upgrades, code changes have been made to use the updated dependencies:
* `OptimaizeLangDetector` is now used in place of the deprecated `LanguageIdentifier`
* The new library versions have removed processing of certain fields so fallback logic has been added to `AttachmentProcessor`
* Attachment Processor unit tests have been updated to accomodate non-deterministic results across library upgrades

### Issues Resolved
Once this is merged, the dependabot PR #2138 can be closed
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kartg** [comment]
start gradle check

**Comment by peterzhuamazon** [comment]
start gradle check

**Review by dblock — APPROVED** [review]
LGTM assuming you can get it to green

**Comment by dblock** [comment]
Test failures look legit.

```

org.opensearch.ingest.attachment.AttachmentProcessorTests > testEnglishTextDocument FAILED
    java.lang.IllegalStateException: No language detectors available
        at __randomizedtesting.SeedInfo.seed([346CD47C4FE8213C:CE265966D8557F82]:0)
        at org.apache.tika.language.detect.LanguageDetector.getDefaultLanguageDetector(LanguageDetector.java:67)
        at org.opensearch.ingest.attachment.AttachmentProcessor.execute(AttachmentProcessor.java:138)
        at org.opensearch.ingest.attachment.AttachmentProcessorTests.parseDocument(AttachmentProcessorTests.java:333)
        at org.opensearch.ingest.attachment.AttachmentProcessorTests.parseDocument(AttachmentProcessorTests.java:323)
        at org.opensearch.ingest.attachment.AttachmentProcessorTests.testEnglishTextDocument(AttachmentProcessorTests.java:85)```

**Comment by kartg** [comment]
> Test failures look legit.

Yup, need to figure out how to configure the language detectors. Flipping to draft PR.

**Comment by kartg** [comment]
More failures around non-determinism of language recognition - this time in `IngestAttachmentClientYamlTestSuiteIT`. Repro commands:

```
./gradlew ':plugins:ingest-attachment:yamlRestTest' --tests "org.opensearch.ingest.attachment.IngestAttachmentClientYamlTestSuiteIT" -Dtests.method="test {yaml=ingest_attachment/30_files_supported/Test ingest attachment processor with .doc file}" -Dtests.seed=1E07D222039A022B -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-ES -Dtests.timezone=CNT -Druntime.java=17

./gradlew ':plugins:ingest-attachment:yamlRestTest' --tests "org.opensearch.ingest.attachment.IngestAttachmentClientYamlTestSuiteIT" -Dtests.method="test {yaml=ingest_attachment/30_files_supported/Test ingest attachment processor with .docx file}" -Dtests.seed=1E07D222039A022B -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-ES -Dtests.timezone=CNT -Druntime.java=17
```

Looks like the failure are limited to doc/docx file processing, and the assertion failure is the same in all cases:

```
expected String [pl] but was String [en]
```

**Comment by kartg** [comment]
This is an error with the test case, and the new output is more accurate.

The two files are represented as Base64 encoded strings [in the yml test file](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/30_files_supported.yml). Decoding them and opening as Word documents shows the contents to be:

```
Test opensearch
```

Previously, language detection was incorrectly identifying this as Polish (`pl`) - likely due to the phrase being so short. With the upgraded libraries, the result has changed to `en` which is an accurate value

**Review by mch2 — APPROVED** [review]

File: plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/AttachmentProcessor.java [hunk 1]
```diff
@@ -134,9 +135,10 @@ public IngestDocument execute(IngestDocument ingestDocument) {
         }
 
         if (properties.contains(Property.LANGUAGE) && Strings.hasLength(parsedContent)) {
-            // TODO: stop using LanguageIdentifier...
-            LanguageIdentifier identifier = new LanguageIdentifier(parsedContent);
-            String language = identifier.getLanguage();
+            OptimaizeLangDetector langDetector = new OptimaizeLangDetector();
```
Comment (reta) [inline plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/AttachmentProcessor.java:138]: It looks like initialization could be an expensive operation, should `OptimaizeLangDetector` + `loadModels()` be done only once during `AttachmentProcessor` initialization?

**Comment by owaiskazi19** [comment]
@kartg should we backport this to 2.x?

**Comment by kartg** [comment]
@owaiskazi19 Yes, looks like it is blocking [this backport](https://github.com/opensearch-project/OpenSearch/pull/3248#issuecomment-1122592793).

I'll dig into @reta 's comment above and make sure any fixes for that get backported too.

**Pull request merged**